lib/src/fnc/util/http/mod.rs

```diff
@@ -6,6 +6,10 @@ use crate::sql::value::Value;
 use surf::Client;
 use surf::Config;
 
+pub(crate) fn uri_is_valid(uri: &str) -> bool {
+	surf::Url::parse(uri).is_ok()
```

Comment:
I did check the `surf` API for non-panicking algernatives to `get`, `post`, etc. but found none, so we're stuck parsing the URI before `surf` does for validation.

Reply:
@finnbear out of interest is there any reason this is `pub(crate)`?

Reply:
> @finnbear out of interest is there any reason this is `pub(crate)`?

IMO everything in the file should be `pub(crate)` because it is a utility and not part of the SurrealDB lib's public, backwards compatible API. However, I kept the diff to a minimum by not modifying other functions.

Reply:
Oh hang on I'm being stupid - it's a separate module 😖

Reply:
I meant 'why is it not private', but I didn't see it was a separate file 😀

Reply:
Oh, that makes a lot of sense :+1:

Reply:
While we're at it then, should we make the other functions in this file `pub(crate)`?

Reply:
I'd be happy to edit as much as possible to `pub(crate)` in a new PR. IMO it would add a lot of noise to this PR, though.